File: 프로그래머스/lv2/12953. N개의 최소공배수/N개의 최소공배수.java
class Solution {
    
    // 5. 최대공약수 메소드, 유클리드 호제법
    public static int gcd(int a, int b){
        int x = Math.max(a,b);
        int y = Math.min(a,b);
        
        while(x % y != 0){
            int r = x % y;
            x = y;
            y = r;
        }
        
        return y;
    }
    
    public int solution(int[] arr) {
        // 1. 배열의 첫 번째 값을 저장.
        int answer = arr[0];
        
        // 2. 배열의 길이가 1이면 반복x, 그 이상은 다음 인덱스인 1부터 반복.
        for(int i = 1;i<arr.length;i++){
            // 3. 두 값의 최대공약수
            int gcd = gcd(answer,arr[i]);
            // 4. 두 값의 최소공배수
            answer = answer * arr[i] / gcd;
        }
        
        return answer;
    }
}

File: 프로그래머스/2/12953. N개의 최소공배수/N개의 최소공배수.py
def solution(arr):
    def GCD(x,y):
        while y:
            x,y = y,x%y
        return x
    
    def LCM(x,y):
        result = x*y//GCD(x,y)
        return result
    
    result = LCM(arr[0],arr[1])
    for i in range(2,len(arr)):
        result = LCM(result,arr[i])
    
    return result